src/XrdHttp/XrdHttpProtocol.cc

```diff
@@ -652,6 +657,20 @@ int XrdHttpProtocol::Process(XrdLink *lp) // We ignore the argument here
           TRACE(DEBUG, " Parsing of first line failed with " << result);
           return -1;
         }
+
+
+        // We permit TLS client auth to be deferred until after the request path is sent.
+        // If this is a path requiring client auth, then do that now.
+        if (tlsClientAuth == XrdTlsContext::ClientAuthSetting::kDefer)
+           {for (const auto &prefix : tlsAuthRequestPrefixes) {
+               {if (!strncmp(prefix.c_str(), CurrentReq.resource.c_str(), prefix.length()))
```

Comment:
This looks like it could be pretty slow in practice if there are several prefixes to be checked and every new request forces the server to go through this loop. Also, this seems to have the problem that `/foo` prefix configured would force authentication on `/foobar`, which may not be intended (similar to the discussion we've had with tokens). I'm not sure what better alternative we could have to handle this, but I'd like to have tests in the test suite before merging, to ensure that this bit really works as it's a security issue if we get it wrong.